src/policy/mod.rs

```diff
@@ -0,0 +1,389 @@
+use std::collections::HashMap;
+
+use crate::crypto::{Hash, PublicKey};
+
+/// A Sigsum policy.
+///
+/// The Sigsum policy dictates if a signed tree head is considered valid (and by extension, if a
+/// Sigsum signature is valid).
+#[derive(Debug, Eq, PartialEq)]
+pub struct Policy {
+    // logs keeps the list of log keys and URLs indexed by keyhash.
+    logs: HashMap<Hash, Entity>,
+
+    // witnesses keeps the list of witness keys and URLs indexed by keyhash.
+    witnesses: HashMap<Hash, Entity>,
+
+    // quorums keeps the quorum entries (witnesses and groups), indexed by name.
+    quorums: HashMap<String, Quorum>,
+
+    // quorum is the quorum required by the policy. Invariant: if present, quorum must be an
+    // existing key in quorums.
+    quorum: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Log {
+    pub pubkey: PublicKey,
+    pub url: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Witness {
+    pub name: String,
+    pub pubkey: PublicKey,
+    pub url: Option<String>,
+}
+
+pub struct Logs<'a> {
+    inner: std::collections::hash_map::Iter<'a, Hash, Entity>,
+}
+
+impl<'a> Iterator for Logs<'a> {
+    type Item = Log;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let (_, entity) = self.inner.next()?;
+        Some(Log {
+            pubkey: entity.0.clone(),
+            url: entity.1.clone(),
+        })
+    }
+}
+
+impl Policy {
+    pub fn logs(&self) -> Logs<'_> {
+        Logs {
+            inner: self.logs.iter(),
+        }
+    }
+
+    pub fn get_witness_by_keyhash(&self, keyhash: &Hash) -> Option<Witness> {
+        let entity = self.witnesses.get(keyhash)?;
+        let name = self
+            .quorums
+            .iter()
+            .filter_map(|(n, q)| match q {
+                Quorum::Witness(h) if h == keyhash => Some(n),
+                _ => None,
+            })
+            .next()
+            .unwrap();
+        Some(Witness {
+            name: name.clone(),
+            pubkey: entity.0.clone(),
+            url: entity.1.clone(),
+        })
+    }
+
+    pub fn quorum(&self) -> Option<String> {
+        todo!()
```

Comment:
Todo?

Reply:
I removed it for now. Implementing this will depend on how the low-level API is implemented.